lib/pubid_iso/document.rb

```diff
@@ -0,0 +1,57 @@
+require "algolia"
+
+module PubidIso
+  class Document
+    attr_accessor :code, :url
+
+    def initialize(code)
+      @code = code
+      @url = fetch_iso.first[:path]
+    end
+
+    def self.parse(code)
+      new(code: Parser.new.parse(code).to_s)
+    rescue Parslet::ParseFailed => failure
+      raise PubidIso::Errors::ParseError, "#{failure.message}\ncause: #{failure.parse_failure_cause.ascii_tree}"
+    end
+
+    def fetch_iso # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
```

Comment:
Obtaining the identifiers from the ISO web site is not really important. The only important thing is to support the bulk list I provided in #1 . ISO will provide us a bulk list later.

Reply:
> Obtaining the identifiers from the ISO web site is not really important. The only important thing is to support the bulk list I provided in #1 . ISO will provide us a bulk list later.

There is not enough data only from Algolia, it's returning attributes:
`:category, :path, :uuid, :title, :status, :year, :order, :text, :objectID, :_highlightResult`
But I need more data to build URN, such as: `stage, edition, languages, year`
Using scrapper, just by having identifier "ISO 4" I can get URN like "urn:iso:std:iso:4:stage-90.93:ed-3:en,fr"

Reply:
The PubID scheme is separate from the URN scheme, not all elements in the URN are available in a PubID. For example, the "edition" is not available (in URN, it is the edition number e.g. `ed-3`. In PubID, it is the year e.g. `1999`, and they cannot be converted into one another).

The stage codes are defined, e.g. WD is 20.00, CD is 30.00, DIS is 40.00, FDIS is 50.00, published (no stage code) means 60.60.

The language code is specified in the identifier. e.g. "ISO 9001:2005(E)" means English, "ISO 9001:2005(F)" is French, "ISO 9001:2005(R)" is Russian.

Reply:
I don't think we need Algolia in general. The URN work should be refactored into a separate class, too.

Reply:
> The stage codes are defined, e.g. WD is 20.00, CD is 30.00, DIS is 40.00, FDIS is 50.00, published (no stage code) means 60.60.

It's not entirely clear to me, because we have more than just 20.00, 30.00, 40.00, 50.00 and 60.60 stages.
For example "ISO 10142:1996" at the stage 90.93. We can't get this from PubID, if we assume that no stage code == 60.60 it's wrong for this case.